### PR TITLE
new task types, resource table and API for Snap Deals

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
Signed-off-by: Alexey Chernyshov <alexey.n.chernyshov@gmail.com>

### Description of the Change
New Task Types, resource table tfor snap deals and worker API changes.

### CODE REVIEW

**Review by turuslan — APPROVED** [review]

File: core/primitives/resources/resources.hpp [hunk 1]
```diff
@@ -297,10 +296,140 @@ namespace fc::primitives {
...
+                    .threads = 1,
+                    .can_gpu = true,
+                    .base_min_memory = uint64_t(2) << 10,
+                }},
+           }},
       };
 
       res[kTTUnseal] = res[kTTPreCommit1];
       res[kTTReadUnsealed] = res[kTTFetch];
+      res[kTTRegenSectorKey] = res[kTTReplicaUpdate];
```
Comment (ortyomka) [inline core/primitives/resources/resources.hpp:432]:
```suggestion
      res[kTTReadUnsealed] = res[kTTFetch];
      res[kTTReplicaUpdate] = res[kTTAddPiece];
      res[kTTProveReplicaUpdate1] = res[kTTCommit1];
      res[TTProveReplicaUpdate2] = res[kTTCommit2];
      res[kTTRegenSectorKey] = res[kTTReplicaUpdate];
```

File: core/remote_worker/remote_worker_api.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    };
+
+    worker_api->Info = [&]() { return worker->getInfo(); };
+    worker_api->Paths = [&]() { return worker->getAccessiblePaths(); };
+    worker_api->TaskTypes =
+        [&]() -> outcome::result<std::set<primitives::TaskType>> {
+      OUTCOME_TRY(tasks, worker->getSupportedTask());
+      // TODO(ortyomka): [FIL-344] Remove its
+      tasks.extract(primitives::kTTAddPiece);
+      return std::move(tasks);
```
Comment (ortyomka) [inline core/remote_worker/remote_worker_api.cpp:58]:
```suggestion
      return  worker->getSupportedTask();
```

File: core/remote_worker/remote_worker_api.cpp [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    };
+    worker_api->SealCommit2 = [&](const SectorRef &sector,
+                                  const Commit1Output &commit_1_output) {
+      return worker->sealCommit2(sector, commit_1_output);
+    };
+    worker_api->FinalizeSector = [&](const SectorRef &sector,
+                                     std::vector<Range> keep_unsealed) {
+      return worker->finalizeSector(sector, keep_unsealed);
+    };
+
```
Comment (ortyomka) [inline core/remote_worker/remote_worker_api.cpp:86]: Please add ReplicaUpdate, ProveReplicaUpdate1 and ProveReplicaUpdate2

Comment (Alexey-N-Chernyshov) [reply core/remote_worker/remote_worker_api.cpp:86]: done

**Review by ortyomka — APPROVED** [review]

**Pull request merged**